src/builtins/strings.rs

```diff
@@ -146,11 +146,32 @@ fn split(span: &Span, params: &[Ref<Expr>], args: &[Value], _strict: bool) -> Re
     let s = ensure_string(name, &params[0], &args[0])?;
     let delimiter = ensure_string(name, &params[1], &args[1])?;
 
-    Ok(Value::from_array(
-        s.split(delimiter.as_ref())
-            .map(|s| Value::String(s.into()))
-            .collect(),
-    ))
+    let mut parts: Vec<Value> = s
+        .split(delimiter.as_ref())
+        .map(|s| Value::String(s.into()))
+        .collect();
+
+    // Handle https://github.com/microsoft/regorus/issues/291
+    // If delimiter is "", rust returns a leading and trailing "", Golang doesn't.
+    if delimiter.as_ref() == "" {
+        if parts.len() >= 2 {
```

Comment:
Any chance this could be simplified, and just set parts to `s.chars().collect()` in this case?

Reply:
Great suggestion. Fixed.